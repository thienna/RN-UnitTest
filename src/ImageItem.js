import React from 'react';
import { ImageBackground, StyleSheet, Text } from 'react-native';
import Colors from 'react-native/Libraries/NewAppScreen/components/Colors';

class ImageItem extends React.PureComponent {
  render() {
    return (
      <ImageBackground
        accessibilityRole={'image'}
        source={require('../node_modules/react-native/Libraries/NewAppScreen/components/logo.png')}
        style={styles.background}
        imageStyle={styles.logo}>
        <Text style={styles.text}>{this.props.text}</Text>
      </ImageBackground>
    );
  }
}

const styles = StyleSheet.create({
  background: {
    paddingBottom: 40,
    paddingTop: 96,
    paddingHorizontal: 32,
    backgroundColor: Colors.lighter,
  },
  logo: {
    opacity: 0.2,
    resizeMode: 'contain',
    backgroundColor: 'red',
    /*
     * These negative margins allow the image to be offset similarly across screen sizes and component sizes.
     *
     * The source logo.png image is 512x512px, so as such, these margins attempt to be relative to the
     * source image's size.
     */
    // marginLeft: -128,
    // marginBottom: -192,
  },
  text: {
    fontSize: 40,
    fontWeight: '600',
    textAlign: 'center',
    color: Colors.black,
  },
});

export default ImageItem;
