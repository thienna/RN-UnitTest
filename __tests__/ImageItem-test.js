/**
 * @format
 */

import 'react-native';
import React from 'react';
import ImageItem from '../src/ImageItem';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Testing ImageItem component', () => {
  it('renders image correctly', () => {
    const wrapper = shallow(
      <ImageItem text={'Sun'}/>,
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ text: 'Sun2' });
    expect(wrapper).toMatchSnapshot();
  });
});
