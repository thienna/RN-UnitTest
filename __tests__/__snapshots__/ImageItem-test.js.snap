// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ImageItem component renders image correctly 1`] = `ShallowWrapper {}`;

exports[`Testing ImageItem component renders image correctly 2`] = `ShallowWrapper {}`;

exports[`Testing ImageItem component renders image3 correctly 1`] = `ShallowWrapper {}`;

exports[`Testing ImageItem component renders image3 correctly 2`] = `ShallowWrapper {}`;
